src/ja/common/job.py

```diff
@@ -0,0 +1,137 @@
+"""
+This module contains the definition of the Job class and related structures.
+"""
+from typing import List
+from enum import Enum
+from ja.common.docker_context import DockerConstraints, IDockerContext
+
+
+class JobStatus(Enum):
+    """
+    Represents the status of a job.
+    """
+    NEW = 0
+    QUEUED = 1
+    RUNNING = 2
+    PAUSED = 3
+    DONE = 4
+    KILLED = 5
+    CRASHED = 6
+
+
+class JobPriority(Enum):
+    """
+    Represents the base priority of a job.
+    """
+    LOW = 0
+    MEDIUM = 1
+    HIGH = 2
+    URGENT = 3
+
+
+class JobSchedulingConstraints:
```

Comment:
How are you deciding which parts of the Job class to put in this class?
It seems kind of arbitrary.
I could argue that a Job's cpu_threads and memory affect "how it is scheduled".

Reply:
I split them by what the constraints apply to. RAM and CPU are constraints for the work machine, whereas priority, preemption, etc. are related purely to the scheduling.

The reason why I split them in two parts is purely aesthetic: I am not a fan of constructors or any method which take a lot of paramaters.


I could merge them together though if you are against the split.